### DISCUSSION TITLE
Getting the exact runtime version from DLLs

### DISCUSSION THREAD

**Discussion by scharnyw** [dotnet/runtime#64321] [Q&A] [answered]
I noticed that since .NET 5, most assemblies in the runtime have their assembly versions set as the major version of the runtime, but the patch version information is lost. For example, from runtime version 5.0.0 through 5.0.13, the assembly version for `System.Net.Http.dll` stays at 5.0.0.0, even though the assembly content has changed due to the patches applied.

This makes it difficult to diagnose potential issues that behave differently across different patch versions of the runtime, because very often I need to backtrack the runtime version from the DLLs. The only reliable way I found so far is to record the assembly MVID of the DLL, and then comparing it against the MVIDs of that assembly in each published patch version of the runtime, which as you may imagine is very time-consuming. Specifically, I'm wondering:

1. Why isn't the assembly versions of the runtime set to be consistent with the runtime version?
2. Is there any way to tell the exact runtime version (e.g. 5.0.13) given a runtime DLL?

Thanks a lot for any help you can provide!

**Comment by PathogenDavid** [comment, marked as answer]
You can get the full version of a given runtime DLL by looking at the `AssemblyInformationalVersionAttribute`.

On Windows you can view it by right-clicking the file in Explorer, opening `Properties`, and then go to the `Details` tab:

![image](https://user-images.githubusercontent.com/278957/151145358-5bff5062-daff-4872-b6d0-30b457e6b30a.png)

**Reply by scharnyw** [reply]
Ah, I missed this attribute when looking at the metadata. Turns out that `System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription` also looks for this attribute on `System.Private.CoreLib.dll`, so this is certainly the definitive way to get the version. Thanks a lot for the help!
